ui/src/app/services/profile-storage.service.ts

```diff
@@ -0,0 +1,27 @@
+import 'rxjs/Rx';
+
+import {Injectable} from '@angular/core';
+import {ReplaySubject} from 'rxjs/ReplaySubject';
+
+import {Profile} from 'generated';
+import {ProfileService} from 'generated';
+
+
+@Injectable()
+export class ProfileStorageService {
+  private activeCall = false;
+  private profile = new ReplaySubject<Profile>(1);
+  public profile$ = this.profile.asObservable();
+
+  constructor(private profileService: ProfileService) {}
+
+  requestNewProfile() {
```

Comment:
Perhaps name this `reload()`, then (maybe) dispatch the initial `getMe()` on service creation (possibly blocked on signin)? This would allow us to satisfy components that would just take a stale view of the profile, as well as those that want to force a reload. e.g. the bug report button probably shouldn't need to force a Profile reload.

Reply:
Sounds good! Will do.